File: python/dijkstra.py
import sys
import time
import json


def dijkstra( punkt_start, punkt_koniec, graf):

    start_time = time.time();
    
    max = 99999;
    wielkosc = len(graf);
    wynik = [];

    dystans=[];
    for i in range(wielkosc):
      dystans.append(max);
    
    dystans[punkt_start] = 0;
      
    tab_wizytacji=[];    
    for i in range(wielkosc):
      tab_wizytacji.append(False);
    
    for i in range(wielkosc):
      for j in range(wielkosc):
        if graf[i][j] == 0:
          graf[i][j] = max;
          
    for g in range(wielkosc):
      wynik.append([]);
      for k in range(wielkosc):
        wynik[g].append(0);
        
    for g in range(wielkosc):
      punkt_start = g;
      for k in range(wielkosc): 
        punkt_koniec = k;
        
        for i in range(wielkosc):
          dystans[i] = max;
        
        dystans[punkt_start] = 0;
             
        for i in range(wielkosc):
          tab_wizytacji[i] = False;
      
        for i in range(wielkosc):      
          aktualny = -1;
          for j in range(wielkosc): 
            if tab_wizytacji[j]:
                continue;
            elif aktualny == -1 or dystans[j] < dystans[aktualny]:
              aktualny = j;
          
          tab_wizytacji[aktualny] = True;
          
          for j in range(wielkosc): 
            droga = dystans[aktualny] + graf[aktualny][j];
            if droga < dystans[j]:
              dystans[j] = droga;
              
        wynik[k][g]=dystans[k];
        wynik[g][k]=dystans[k];
		
    elapsed_time = time.time() - start_time;
    elapsed_time *= 1000;
    print("%.0f" % (elapsed_time));
    return wynik;


size = 10;
arg = sys.argv[1];

if arg == "10":
    size=10;
elif arg == "100":
    size=100;
elif arg == "200":
    size=200;
elif arg == "300":
    size=300;

name= '%sx%s' % (size, size)
path = "../grafy/" + name + "/graf_" + name + ".json";


with open(path) as f:
    data = json.load(f) # zwraca liste

graf = data;
	
#print(graf)

wynik = dijkstra(1,1,graf)
#print(wynik[4][3]);

File: java/dijkstra.java
public class dijkstra {

	public static void main(String[] args) {
			
			
			int[][] graf = {{ 0, 0, 0, 1, 90, 35, 89, 2, 0, 28},{ 0, 0, 1, 19, 0, 0, 0, 0, 2, 85},{ 0, 1, 0, 0, 7, 0, 0, 0, 0, 0},{ 1, 19, 0, 0, 0, 0, 91, 0, 0, 0},{ 90, 0, 7, 0, 0, 0, 78, 0, 15, 61},{ 35, 0, 0, 0, 0, 0, 0, 0, 81, 0},{ 89, 0, 0, 91, 78, 0, 0, 0, 0, 0},{ 2, 0, 0, 0, 0, 0, 0, 0, 0, 3},{ 0, 2, 0, 0, 15, 81, 0, 0, 0, 0},{ 28, 85, 0, 0, 61, 0, 0, 3, 0, 0}};


			int[][] wynik = algorytm(1,2,graf); 
			
			System.out.println(wynik[4][2]);
		
	}
	
	
	static int[][] algorytm( int punkt_start, int punkt_koniec, int[][] graf)
	{

		long startTime = System.currentTimeMillis();
		int wielkosc = graf.length;

		int[] dystans = new int[wielkosc]; 
		boolean[] tab_wizytacji = new boolean[wielkosc]; 
		
		int[][] wynik = new int[wielkosc][wielkosc]; 

		for (int g = 0; g<wielkosc; ++g) {
			for (int k = 0; k<wielkosc; ++k) {
				wynik[k][g] = 0;
			}
		}
		
		int max = 999;
		for(int i = 0; i < wielkosc; ++i){

	        for(int j = 0; j < wielkosc; ++j)
	        {
	            if(graf[i][j] == 0){
	                graf[i][j] = max;
	            }
	        } 
	    }
		for (int g = 0; g<wielkosc; ++g) {
			punkt_start = g;
			for (int k = 0; k<wielkosc; ++k) {
				punkt_koniec = k;
		
				dystans = new int[wielkosc]; 
				tab_wizytacji = new boolean[wielkosc]; 
		
				
		
				for(int i = 0; i < wielkosc; ++i) {
				    dystans[i] = max; // wielkosc maksymalna
				}
				dystans[punkt_start] = 0;
		
				for(int i = 0; i < wielkosc; ++i) {
				    tab_wizytacji[i] = false;
				}
		
				for(int i = 0; i < wielkosc; ++i){
			        int aktualny = -1;
			        for(int j = 0; j < wielkosc; ++j){
			            if(tab_wizytacji[j])
			            {
			                continue;
			            }
			            if(aktualny == -1 || dystans[j] < dystans[aktualny])
			            {
			                aktualny = j;
			            }
			        }
			        
			        tab_wizytacji[aktualny] = true;
		
			        for(int j = 0; j < wielkosc; ++j)
			        {
			            int droga = dystans[aktualny] + graf[aktualny][j];
			            if(droga < dystans[j]){
			                dystans[j] = droga;
			            }
			        }  
			    }
				
				wynik[k][g] = dystans[k];
				wynik[g][k] = dystans[k];
		
			}
	}


		long estimatedTime = System.currentTimeMillis() - startTime;
		System.out.println("Czas " + estimatedTime + " ms.");  

		return wynik;
		
	}

}
